cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/keyspaces/KeyspaceCreator.java

```diff
@@ -83,38 +87,43 @@ public void createKeyspace() throws IOException, CradleStorageException
 		Optional<KeyspaceMetadata> meta = obtainKeyspaceMetadata();
 		if (meta.isPresent())
 			throw new CradleStorageException("Keyspace '" + keyspace + "' already exists");
-
+		
 		logger.info("Creating keyspace '{}'", keyspace);
 		CreateKeyspace createKs = settings.getNetworkTopologyStrategy() != null
-				? SchemaBuilder.createKeyspace(keyspace).withNetworkTopologyStrategy(settings.getNetworkTopologyStrategy().asMap())
+				? SchemaBuilder.createKeyspace(keyspace)
+				.withNetworkTopologyStrategy(settings.getNetworkTopologyStrategy().asMap())
 				: SchemaBuilder.createKeyspace(keyspace).withSimpleStrategy(settings.getKeyspaceReplicationFactor());
 		queryExecutor.executeQuery(createKs.asCql(), true);
 		logger.info("Keyspace '{}' has been created", keyspace);
-
+		
 		awaitKeyspaceReady();
 	}
-
+	
 	private void awaitKeyspaceReady() throws CradleStorageException
 	{
-		int timeout = 500;
 		int attempt = 0;
-		for (;getKeyspaceMetadata() == null && attempt < 5; attempt++)
+		for (; getKeyspaceMetadata() == null && attempt < ATTEMPTS_KEYSPACE_WAITING; attempt++)
 		{
+			logger.debug("[{}] attempt to wait {}ms for the readiness of the keyspace", attempt + 1, keyspaceWaitTimeout);
 			try
 			{
-				TimeUnit.MILLISECONDS.sleep(timeout);
+				TimeUnit.MILLISECONDS.sleep(keyspaceWaitTimeout);
 			}
 			catch (InterruptedException e)
 			{
-				throw new CradleStorageException("Waiting for keyspace '"+keyspace+"' readiness has been interrupted", e);
+				throw new CradleStorageException(
+						"Waiting for keyspace '" + keyspace + "' readiness has been interrupted", e);
 			}
 		}
-
+		
 		if (getKeyspaceMetadata() == null)
 			throw new CradleStorageException(
-					"Keyspace '" + keyspace + "' unavailable after " + attempt * timeout + "ms of awaiting");
+					"Keyspace '" + keyspace + "' unavailable after " + attempt * keyspaceWaitTimeout +
+							"ms of awaiting");
+		
+		logger.debug("Keyspace '{}' was ready in {}ms after creation", keyspace, attempt * keyspaceWaitTimeout);
```

Comment:
This looks like keyspace creation took `attempt * keyspaceWaitTimeout` while actually it could take less.